proxy/src/main/java/net/md_5/bungee/connection/UpstreamBridge.java

```diff
@@ -131,7 +130,11 @@ public void handle(KeepAlive alive) throws Exception
 
         if ( keepAliveData != null && alive.getRandomId() == keepAliveData.getId() )
         {
-            Preconditions.checkState( keepAliveData == con.getServer().getKeepAlives().poll(), "keepalive queue mismatch" );
+            if ( keepAliveData != con.getServer().getKeepAlives().poll() )
+            {
+                con.disconnect( "keepalive queue mismatch" );
```

Comment:
perhaps explain to users what to do?

Reply:
What the user has to do? Behave like a vanilla client.

Reply:
idk, this doesn’t give much information to the user

Reply:
Now that I look at it again this does more look like a failsafe which should never happen

Reply:
Yes this should never happen so we don't need to give information about it